handsontable/src/utils/parseTable.js

```diff
@@ -59,13 +59,24 @@ export function instanceToHTML(hotInstance) {
  * @param {object} config Configuration for building HTMLTableElement.
  * @param {Array<number>} config.rows List of row indexes which should be taken into account when creating the table.
  * @param {Array<number>} config.columns List of column indexes which should be taken into account when creating the table.
+ * @param {boolean} [config.excelMode] Gets to format handled by desktop Excel application.
  * @returns {string} OuterHTML of the HTMLTableElement.
  */
 export function getHTMLByCoords(hotInstance, config) {
+  const headers = getHeadersHTMLByCoords(hotInstance, config);
+  const body = getBodyHTMLByCoords(hotInstance, config);
+  const excelMode = isDefined(config.excelMode) ? config.excelMode : true;
+  const multipleElements = headers.length + body.length > 5; // Has at least two cells/headers
+  const isAddedCommentForExcel = excelMode && multipleElements;
+
   return [
     '<table>',
-    ...getHeadersHTMLByCoords(hotInstance, config),
-    ...getBodyHTMLByCoords(hotInstance, config),
+    // Needed for desktop Excel on MacOS while pasting any elements with rowspan/colspan.
+    isAddedCommentForExcel ? '<!--StartFragment-->' : '',
+    ...headers,
+    ...body,
+    // Needed for desktop Excel on MacOS while pasting any elements with rowspan/colspan.
+    isAddedCommentForExcel ? '<!--EndFragment-->' : '',
```

Comment:
```suggestion
    // Needed for desktop Excel on MacOS while pasting any elements with rowspan/colspan.
    '<!--StartFragment-->',
    ...headers,
    ...body,
    // Needed for desktop Excel on MacOS while pasting any elements with rowspan/colspan.
    '<!--EndFragment-->',
```
Is there any reason why you have not hardcoded those HTML comments? They are meaningful for Excel but should be ignored by other software as soon they are just comments. Thanks to that, you could simplify the code.

Reply:
1. This method is used internally by `toHTML` method and in e2e tests which shouldn't include comments.
2. Comments are added by Excel only in cases when we have more than one cell in the copied selection.